components/bl-markdown-component/src/index.js

```diff
@@ -0,0 +1,62 @@
+import MarkdownIt from './markdown-it.min.js';
+import { useState, useEffect } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function MyCustomComponent({ component }) {
+  const { classList, style, display, url, markdownText, height, width } = component;
+
+  const [markdown, setMarkdown] = useState('');
+  const [isLoad, setIsLoad] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    const md = new MarkdownIt();
+    if (!markdownText && url) {
+      setIsLoad(true);
+      fetch(url)
+        .then(response => response.text())
+        .then((data) => {
+          setIsLoad(false);
+          setMarkdown(md.render(data));
+        })
+        .catch((error) => {
+          setIsLoad(false);
+          setErrorMessage(error.message);
+        });
+    } else if (markdownText) {
+      setMarkdown(md.render(markdownText));
+    } else {
+      setMarkdown('<h1>No specified markdown text</h1>');
+    }
+  }, [markdownText, url]);
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div
+      className={ cn('bl-customComponent-markdown markdown-body', classList) }
+      style={ { ...style, height: height || '100%', width: width || '100%' } }>
+      { isLoad && <Loader/> }
+
+      { errorMessage && !isLoad ? (
+        <span className="error-message"> { errorMessage } </span>
+      ) : (
+        <div className="markdown-body" dangerouslySetInnerHTML={ { __html: markdown } }/>
+      ) }
```

Comment:
I'm not pretty love notations like this, they look a little dirty.

What about something like that?

```
const renderMdContent = () => {
   if(isLoading) {
     return <Loader>
   }
   
   if(errorMessage) {
     return <Error>
   }

  return <div className="markdown-body" .../>
}
```

And here 

```
return (
  <div>
    { renderMdContent() }
  </div>
)
```

It's just an example, you can refactor it using this principle